### DISCUSSION TITLE
map interface needs a location search

### DISCUSSION THREAD

**Discussion by kamil-malinski** [directus/directus#12633] [Archive]
Hi!

I just tried the map interface and I think it would be much more useful if one could search for an address inside the map interface. Otherwise it is pretty hard to set the pin to the desired location if you just know the address of a location.

Cheers

Kamil

**Comment by rijkvanzanten** [comment]
This functionality already exists 🙂 

You will have to add a Mapbox key into Project Settings to enable it though, as geo-lookup requires third party APIs:

<img width="1006" alt="CleanShot 2022-04-08 at 09 18 36@2x" src="https://user-images.githubusercontent.com/9141017/162443805-1be76b3a-38d4-4f3e-9eb8-45cdccc92f1f.png">

**Reply by kamil-malinski** [reply]
oh, that’s great, I didn’t know that… thanks for letting me know!

**Reply by Resumane** [reply]
Not only MapBox isn't free, but it is total garbage for my use: it can't find any buildings in my city/country. OpenStreetMap, on the other hand, is free and has all the data.
I understand that you probably make some money if people use the MapBox paid API, but it would be so much better if the OpenStreetMap integration had search because it's 100x better than MapBox.

**Reply by rijkvanzanten** [reply]
@Resumane I don't get any kickback from Mapbox. My 3 yo comment still holds true today: geo-location search requires a third party API to do the searching, and as of today we support Mapbox as a search API for that use. If you'd like to see additional geo-lookup APIs to be supported, feel free to open a new feature request 👍🏻 The OSM foundation's free Nominatim lookup service has a strict usage policy and a 1 request per second per application restriction which makes it unsuitable to ship out of the box with for a platform like Directus

**Comment by loongmxbt** [comment]
@rijkvanzanten Hi, I have a similar problem, how can I search a point by lat and lon, e.g. 119.5E, 32.5N. I've tried different combinations like 119,32.5, but still cannot locate to the correct position.

**Reply by azrikahar** [reply]
I believe Mapbox recognizes the reverse, which is `Longitude,Latitude` [as mentioned here](https://docs.mapbox.com/help/glossary/lat-lon/#find-a-locations-coordinates). I tested the combination `-73.98613553935542, 40.7483173708684` over at their [Geocoding API playground](https://docs.mapbox.com/playground/geocoding/) and it works. (Note that popular services like Google Maps uses `Latitude,Longitude` instead, which might be the source of confusion)

**Reply by loongmxbt** [reply]
<img width="848" alt="image" src="https://user-images.githubusercontent.com/5986600/162668443-82025ce2-cf87-41ee-8c1d-d483c3b7d112.png">
Hi, thanks for your reply. In mapbox studio, I can geolocate a destination by lon, lat. But in directus map, when I input lon, lat, seems it did't recognize. It will parse it to some road/avenue number.
Currently I change the map field into type `Json` and use json editor to input the lat and lon.

**Reply by benhaynes** [reply]
That is a geo-lookup field, not sure how that library processes things... but it might not support Lat/long. I believe @azrikahar was talking about our API (not the App interface).

**Reply by rijkvanzanten** [reply]
Also the search box in the map interface is meant to find a lat long based on an address input. I'm not sure if that same API endpoint will return the lat long output for a lat long input 🤔

**Reply by phihu** [reply]
I know it's a pretty old topic, but I had a similar problem and at least there is a somewhat practical workaround:

Just add a marker anywhere on the map and edit the field's raw data JSON (there are only two numbers, the lat/lng in _decimal_ form). 

At least this way you can directly work with/use coordinates.
